diff --git a/ckan/model/meta.py b/ckan/model/meta.py
index 7d677a7d7..0b89c75e2 100644
--- a/ckan/model/meta.py
+++ b/ckan/model/meta.py
@@ -74,8 +74,8 @@ class CkanSessionExtension(SessionExtension):
             session.execute(
                 revision_table.update().where(
                     and_(revision_table.c.id == obj.id,
-                         revision_table.c.current == '1')
-                ).values(current='0')
+                         revision_table.c.current == True)
+                ).values(current=False)
             )

             q = session.query(revision_cls)