diff --git a/ckan/public-bs3/base/javascript/modules/data-viewer.js b/ckan/public-bs3/base/javascript/modules/data-viewer.js
index 5b9cbaf..b86a9ef 100644
--- a/ckan/public/base/javascript/modules/data-viewer.js
+++ b/ckan/public/base/javascript/modules/data-viewer.js
@@ -10,6 +10,7 @@ this.ckan.module('data-viewer', function (jQuery) {

     initialize: function () {
       jQuery.proxyAll(this, /_on/);
+      this._onLoad();
       this.el.on('load', this._onLoad);
       this._FirefoxFix();
       this.sandbox.subscribe('data-viewer-error', this._onDataViewerError);