diff --git a/ckan/config/middleware/pylons_app.py b/ckan/config/middleware/pylons_app.py
index 5a1218d..2644ecd 100644
--- a/ckan/config/middleware/pylons_app.py
+++ b/ckan/config/middleware/pylons_app.py
@@ -1,6 +1,7 @@
 # encoding: utf-8
 
 import os
+import re
 
 from pylons.wsgiapp import PylonsApp
 
@@ -71,6 +72,9 @@ def make_pylons_stack(conf, full_stack=True, static_files=True,
                                     cleanup_pylons_response_string)
 
     # Fanstatic
+    root_path = config.get('ckan.root_path', None)
+    if root_path:
+        root_path = re.sub('/{{LANG}}', '', root_path)
     if asbool(config.get('debug', False)):
         fanstatic_config = {
             'versioning': True,
@@ -78,6 +82,7 @@ def make_pylons_stack(conf, full_stack=True, static_files=True,
             'minified': False,
             'bottom': True,
             'bundle': False,
+            'base_url': root_path,
         }
     else:
         fanstatic_config = {
@@ -86,6 +91,7 @@ def make_pylons_stack(conf, full_stack=True, static_files=True,
             'minified': True,
             'bottom': True,
             'bundle': True,
+            'base_url': root_path,
         }
     app = Fanstatic(app, **fanstatic_config)
 
