diff --git a/ckan/lib/jinja_extensions.py b/ckan/lib/jinja_extensions.py
index ce2608f1e..abedb239d 100644
--- a/ckan/lib/jinja_extensions.py
+++ b/ckan/lib/jinja_extensions.py
@@ -18,8 +18,11 @@ import ckan.lib.helpers as h
 from ckan.common import config
 from markupsafe import Markup

+from werkzeug.local import Local
+

 log = logging.getLogger(__name__)
+local = Local()


 def _get_extensions() -> list[Any]:
@@ -197,12 +200,17 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
             searchpaths = self.searchpath[index + 1:]
         else:
             searchpaths = self.searchpath
+        if not hasattr(local, 'missing_templates'):
+            local.missing_templates = set()
         # end of ckan changes
         pieces = loaders.split_template_path(template)
         for searchpath in searchpaths:
             filename = path.join(searchpath, *pieces)
+            if filename in local.missing_templates:
+                continue
             f = open_if_exists(filename)
             if f is None:
+                local.missing_templates.add(filename)
                 continue
             try:
                 contents = f.read().decode(self.encoding)