diff --git a/ckan/controllers/package.py b/ckan/controllers/package.py
index c8e7dbf..5bd8793 100644
--- a/ckan/controllers/package.py
+++ b/ckan/controllers/package.py
@@ -25,6 +25,8 @@ import ckan.lib.render

 from ckan.common import OrderedDict, _, json, request, c, g, response
 from home import CACHE_PARAMETERS
+from ckan.model.package import Package
+from ckan.lib.dictization.model_dictize import group_list_dictize

 log = logging.getLogger(__name__)

@@ -1228,6 +1230,8 @@ class PackageController(base.BaseController):
         data_dict = {'id': id}
         try:
             c.pkg_dict = get_action('package_show')(context, data_dict)
+            pkg_obj = Package.get(data_dict['id'])
+            c.pkg_dict['groups'] = group_list_dictize(pkg_obj.get_groups('group', None), context)
             dataset_type = c.pkg_dict['type'] or 'dataset'
         except (NotFound, NotAuthorized):
             abort(404, _('Dataset not found'))
@@ -1270,7 +1274,7 @@ class PackageController(base.BaseController):

         c.group_dropdown = [[group['id'], group['display_name']]
                             for group in users_groups if
-                            group['id'] not in pkg_group_ids]
+                            group['id'] not in pkg_group_ids and group['type'] == 'group']

         for group in c.pkg_dict.get('groups', []):
             group['user_member'] = (group['id'] in user_group_ids)
